### ISSUE TITLE
Cannot open folder by clicking on the icon

### ISSUE THREAD

**Issue by AlphaScorpii-dev** [mtkennerly/ludusavi#257]
### Ludusavi version

v0.21.0

### Operating system

Linux (Steam Deck)

### Installation method

Flatpak

### Description

This used to work a couple of months ago, but I honestly cannot say if it’s due to a regression on ludusavi’s side or to a change on the OS side. I can say for sure it wasn’t working anymore in 0.20.

Right now, all the blue icons that appear aside folder names are basically inactive. They appear to be clickable (light change in the display when clicked) but nothing happens.

Before Dolphin was opened on the correct folder without any problem.

**Comment by mtkennerly** [comment]
Hi! Are you trying in desktop mode or game mode? The folder openers don't work in game mode, but they should work in desktop mode.

**Comment by mtkennerly** [comment]
Hmm, I'm not sure then. I can try uploading a beta Flatpak build with some extra logging to help diagnose what's happening.

In the meantime, would you mind trying the [standalone download](https://github.com/mtkennerly/ludusavi/releases/tag/v0.21.0), just to rule out anything with Flatpak specifically?

**Comment by AlphaScorpii-dev** [comment]
I tried with your standalone version, but when running this one the UI scaling is terrible. All the component are so oversized that I only get perhaps 10-20% of the window displayed on my screen.
I remember reading something similar in the other issues, so I need to find it back first.

**Comment by mtkennerly** [comment]
Ah, yeah, you just need to set the `WINIT_X11_SCALE_FACTOR` environment variable to `1`.

**Comment by AlphaScorpii-dev** [comment]
Ok, it works fine, including the folder opening. Seems to be a problem with the flatpak version then.

**Comment by AlphaScorpii-dev** [comment]
I’ve been toying a little with the flatpak permissions in case it would change anything, but unfortunately it didn’t.
So, so far it seems to be a flatpak-only issue, but no workaround with flatseal.

**Comment by mtkennerly** [comment]
For context, Ludusavi uses the [opener](https://crates.io/crates/opener) crate for this. With the version Ludusavi uses (0.5.2), It looks like it first tries to run the system copy of `xdg-open`, and if that fails, then it tries using `sh` to run a bundled copy of `xdg-open`.

I've published a beta Flatpak build that adds some logging and updates the version of `opener`. The new version seems to have added some FreeDesktop-specific support, which could help. Would you mind giving it a try? To install it:

```
flatpak remote-add --if-not-exists flathub-beta https://flathub.org/beta-repo/flathub-beta.flatpakrepo
flatpak install flathub-beta com.github.mtkennerly.ludusavi
flatpak run com.github.mtkennerly.ludusavi//beta
```

The title bar should say "Ludusavi v0.21.0-post.4+1af8e5e". If you see an error message when you try to open a path, then please give me a copy of the log file (`~/.var/app/com.github.mtkennerly.ludusavi/config/ludusavi/ludusavi_rCURRENT.log`).

**Comment by AlphaScorpii-dev** [comment]
It didn’t change anything unfortunately.
Still no folder opened, and no error displayed either. The click just has zero effect.
I have been looking for a log file too but none was generated.

Aug 31, 2023 at 15:13 by ***@***.***:

**Comment by mtkennerly** [comment]
Hmm, that's weird. I've added a little more logging to narrow it down - could you update the beta Flatpak and try one more time? The title bar should say "v0.21.0-post.5+8d41b2e".

**Comment by AlphaScorpii-dev** [comment]
Some log info at last :)
Sorry for the poor formatting as I am replying by email.
Despite the logs saying the folder was opened, nothing happened.
[2023-09-01T12:59:53.017Z] WARN [ludusavi::gui::app] about to open dir: "/home/deck/.local/share/Steam/steamapps/compatdata/1235140/pfx/drive_c/users/steamuser/AppData/Roaming/Sega/YakuzaLikeADragon/xxx"
[2023-09-01T12:59:53.024Z] WARN [ludusavi::gui::app] opened dir: "/home/deck/.local/share/Steam/steamapps/compatdata/1235140/pfx/drive_c/users/steamuser/AppData/Roaming/Sega/YakuzaLikeADragon/xxx", result: Ok(())

**Comment by mtkennerly** [comment]
Could you try launching Ludusavi with the following command?

```
flatpak run --env=XDG_CURRENT_DESKTOP=KDE com.github.mtkennerly.ludusavi//beta
```

I found a couple of threads that indicate the KDE setting might help:

* https://steamcommunity.com/app/1675200/discussions/2/3267932441830585132/
* https://www.reddit.com/r/SteamDeck/comments/u2wofv/steamos_doesnt_come_with_xdgdesktopportalgtk/

**Comment by AlphaScorpii-dev** [comment]
Didn’t work either, sorry.

**Comment by mtkennerly** [comment]
I see; thanks for checking. I'm not sure what else to try :/ In the future, if I get access to a Steam Deck, then I'll try experimenting with it.

**Comment by mtkennerly** [comment]
Actually, yeah, that would be good to verify first. Right now, I only have access to Linux VMs, which don't play nice with Flatpak, but I can try setting up a portable install to test it.

**Comment by AlphaScorpii-dev** [comment]
Is there a way I can make regression test using older versions of the flathub release ? Right now I only have access to the 0.21 release and the beta. As it seems to me it used to work in the past, perhaps it could help tracking down the issue ?

**Comment by mtkennerly** [comment]
It [sounds like](https://discourse.flathub.org/t/install-an-older-version-of-an-app/320/4) you can do this:

* Run `flatpak remote-info --log com.github.mtkennerly.ludusavi`
* From the output, take one of the commit hashes and plug it into this command: `flatpak update --commit=$COMMIT com.github.mtkennerly.ludusavi`

If that doesn't work, let me know and I'll see what I can do.

**Comment by AlphaScorpii-dev** [comment]
Message ID: ***@***.***>I have tried this in a OpenSUSE VM where I installed Flatpak and Ludusavi, for which I tested all the versions down to 0.15.0. So far, none worked.
I guess my memories of it working were from the time when I tested Ludusavi from the first time on my Windows rig...

Interestingly, when I am adding an entry for custom games, clicking on the folder icon does open a Zenity window to help me select a folder. But nothing happens in the preview window in backup mode.

**Comment by AlphaScorpii-dev** [comment]
Message ID: ***@***.***>Sorry, another update... I tried again with a GNOME desktop this time, and it worked with the latest 0.21 version, opening Nautilus.
After uninstalling GNOME and installing Plasma, it still works and opens Dolphin.
Only my first attempt with XFCE didn't work. I guess there is some package missing on the Deck that causes the problem :/

**Comment by AlphaScorpii-dev** [comment]
I tried launching your flatpak with flatpak run --log-session-bus
This generated the following when I tried to click on one folder.

> C1: -> org.freedesktop.DBus call org.freedesktop.DBus.Hello at /org/freedesktop/DBus
> C2: -> org.freedesktop.DBus fake wildcarded AddMatch for org.freedesktop.portal
> C3: -> org.freedesktop.DBus fake wildcarded AddMatch for org.mpris.MediaPlayer2.com.github.mtkennerly.ludusavi
> C4: -> org.freedesktop.DBus fake wildcarded AddMatch for com.github.mtkennerly.ludusavi
> C5: -> org.freedesktop.DBus fake ListNames
> B1: <- org.freedesktop.DBus return from C1
> B2: <- org.freedesktop.DBus signal org.freedesktop.DBus.NameAcquired at /org/freedesktop/DBus
> B3: <- org.freedesktop.DBus return from C2
> *SKIPPED*
> B4: <- org.freedesktop.DBus return from C3
> *SKIPPED*
> B5: <- org.freedesktop.DBus return from C4
> *SKIPPED*
> B6: <- org.freedesktop.DBus return from C5
> C6: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.DBus
> C7: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Desktop
> C8: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Documents
> C9: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.IBus
> *SKIPPED*
> B7: <- org.freedesktop.DBus return from C6
> *SKIPPED*
> B8: <- org.freedesktop.DBus return from C7
> *SKIPPED*
> B9: <- org.freedesktop.DBus return from C8
> *SKIPPED*
> B10: <- org.freedesktop.DBus return from C9
> *SKIPPED*
> C10: -> org.freedesktop.portal.Desktop call org.freedesktop.portal.OpenURI.OpenFile at /org/freedesktop/portal/desktop
> B11: <- org.freedesktop.DBus signal org.freedesktop.DBus.NameOwnerChanged at /org/freedesktop/DBus
> B12: <- org.freedesktop.DBus return error org.freedesktop.DBus.Error.NoReply from C10

**Comment by AlphaScorpii-dev** [comment]
Just to let you know, I think I have cornered the issue, and you're not at fault.
When running flatpak run --command=xdg-open com.github.mtkennerly.ludusavi file:///home/whatever
I could not get any window to pop up.
I thought I had an issue with the deck's default OS installation (missing flatpak-xdg-utils for instance), or that I needed to have a file explorer as a flatpak as well for it to work, you name it...
After looking at mimeinfo.cache, I realized that another flatpak application (filelight) was overriding inode/directory.
Long story short, I deleted it, and ludusavi can now open folders again.
Honestly, I'm not sure why it was failing, but it seems that as soon as one flatpak app overrides inode/directory, file opening stops working (even if it's dolphin...)

**Comment by mtkennerly** [comment]
That's a great find! And quite obscure 😅 Thank you for reporting back with your findings.
